workspaces/diff-engine/src/projections/history.rs

```diff
@@ -68,33 +68,35 @@ impl HistoryProjection {
     parent_batch_id: Option<CommitId>,
     commit_message: String,
   ) {
-    // TODO: refactor how we determine what the root commit is and how it gets created,
-    // perhaps by wrapping the events from specification.json in batch commit, the first
-    // of which has "root" an id. That's best done in the SpecChunk logic.
-    if let Some(parent_batch_id) = parent_batch_id {
-      let node = Node::BatchCommit(BatchCommitNode(
-        batch_id.clone(),
-        BatchCommitDescriptor {
-          commit_message,
-          is_complete: false,
-        },
-      ));
-      let node_index = self.graph.add_node(node);
-      self.node_id_to_index.insert(batch_id.clone(), node_index);
-
-      let parent_node_index = *self
-        .node_id_to_index
-        .get(&parent_batch_id)
-        .expect("expected parent commit to have a corresponding node");
-
+    let parent_batch_id = parent_batch_id.unwrap_or_else(|| {
       self
-        .graph
-        .add_edge(parent_node_index, node_index, Edge::IsParentOf);
+        .get_commit_id(&self.find_last_batch_commit_index().unwrap())
+        .unwrap()
+        .clone()
+    });
```

Comment:
This is an essential change to fix a bug I observed while testing the migration of single-file to multi-file spec. With it, the batch commits inside of `specification.json` also become part of this graph, which in turn allows batch chunks from other files to refer to these batch commits (instead of only being able to refer to the specification file globally, which is what the bug was). The "root" commit, for the purposes of this projection,  becomes basically every event before a first batch commit is observed. 

The added bonus of having the batches embedded in a specification.json part of this projection, is that they can now be walked and rendered in any views of this as well.